### DISCUSSION TITLE
How do I change this colour?

### DISCUSSION THREAD

**Discussion by Hund** [davatorium/rofi#1292] [Q&A] [answered]
How do I change this colour?

![rofitext](https://user-images.githubusercontent.com/320498/112303822-09606b00-8c9d-11eb-8c89-141b4b170d98.png)

**Comment by DaveDavenport** [comment, marked as answer]
The placeholder is:

**• placeholder-color: Color of the placeholder text.**

on the `entry` widget.

the other three are `text-color` on those widgets. (`num-rows`, `num-filtered-rows` etc.)
Please see theme for exact widget name.

Please see rofi-theme manpage for this info.

**Reply by Hund** [reply]
I added this, but it didnd't do anything?

```
placeholder-color: "#000000";
text-color: "#000000";
```

**Reply by DaveDavenport** [reply]
add it where?  this is not enough information.

**Reply by Hund** [reply]
To my config, what else?

`~/.config/rofi/config.rasi`

**Reply by DaveDavenport** [reply]
you need to override the right part of the theme.
Just adding it randomly to the configuration file (still no idea in what section you added?) won't work.

**Reply by Hund** [reply]
Sorry. I got confused. I had forgot that there's settings in two different configs now.

I added this to my `~/.Xresources`:

```
rofi.placeholder-color: #000000
rofi.text-color: #000000
```

It didn't work either.

**Reply by DaveDavenport** [reply]
yes, that won't work, you need to override the right section in the theme. 

maybe pastebin your theme? because I cannot answer without some information.

**Reply by Hund** [reply]
This is my theme:

```
rofi.font: xos4Terminus 12
rofi.bw: 3
rofi.m: -1

rofi.color-enabled: true
rofi.color-window: #002b36, #255662, #0d4350
rofi.color-normal: #002b36, #839496, #002b36, #0d4350, #839496
rofi.color-active: #0d4350, #2aa198, #002b36, #839496, #002B36
rofi.color-urgent: #a83838, #ffffff, #eee8d5, #a83838, #ffffff
rofi.placeholder-color: #000000
rofi.text-color: #000000
```

**Reply by DaveDavenport** [reply]
can you pastebin `rofi -dump-theme`, because that is a long long long long long long long long deprecated format.

**Reply by Hund** [reply]
Sure.

https://dpaste.org/uWjQ/slim

**Reply by DaveDavenport** [reply]
append do your  rasi file

```css
num-filtered-rows {
    text-color: rgba ( 0, 0, 0, 100 % );
}
num-rows {
    text-color: rgba ( 0, 0 0, 100 % );
}
textbox-num-sep {
    text-color: rgba ( 0, 0, 0, 100 % );
}
entry {
    placeholder-color:       "#000000;
}
```
not sure this will work 100% given the old format you use.

**Reply by DaveDavenport** [reply]
if you use new format you can also just remove the placeholder and extra widgets.

**Reply by Hund** [reply]
I'm trying to use this new format, but it doesn't work. I moved the dumped theme to `~/.config/rofi/themes/solarized.rasi` and then pointed to it in the config: `theme: solarized;`, but that didn't work?

I read the manual, but it doesn't mention anything about this.

**Reply by Hund** [reply]
I needed to put the name in qutoes.

**Reply by DaveDavenport** [reply]
can you pastebin the output of `rofi --help`, so I know where it gets it options from and what version you are running.

**Reply by DaveDavenport** [reply]
the Xresources config option will most likely be removed in the next version, so it is good to migrate to newer format.

**Reply by Hund** [reply]
I removed the concig from Xresources and reloaded it, but it still looks like it reads from it?

https://dpaste.org/5gVb/slim

**Reply by DaveDavenport** [reply]
> I removed the concig from Xresources and reloaded it, but it still looks like it reads from it?
> 
> https://dpaste.org/5gVb/slim

did you properly reload the Xresources file? so the settings are removed? (`xrdb -all -query` to see current state)


about blue line, edit the theme to not have a blue border between input bar and listview.

**Reply by Hund** [reply]
I did. I run: `xrdb -merge ~/.Xresources` and it still shows me this:

```
$ xrdb -all -query
[...]
rofi.bw:	3
rofi.color-active:	#0d4350, #2aa198, #002b36, #839496, #002B36
rofi.color-enabled:	true
rofi.color-normal:	#002b36, #839496, #002b36, #0d4350, #839496
rofi.color-placeholder:	#000000
rofi.color-text:	#000000
rofi.color-urgent:	#a83838, #ffffff, #eee8d5, #a83838, #ffffff
rofi.color-window:	#002b36, #255662, #0d4350
rofi.font:	xos4Terminus 12
rofi.m:	-1
rofi.placeholder-color:	#000000
rofi.text-color:	#000000
```

**Reply by Hund** [reply]
I can't hide the scrollbar though. I added `enabled: false;` to this part:

```
scrollbar {
    enabled: false;
}
```

It didnt' work  though? What am I doing wrong?

**Reply by DaveDavenport** [reply]
> I did. I run: `xrdb -merge ~/.Xresources` and it still shows me this:
> 
> ```
> $ xrdb -all -query
> [...]
> rofi.bw:	3
> rofi.color-active:	#0d4350, #2aa198, #002b36, #839496, #002B36
> rofi.color-enabled:	true
> rofi.color-normal:	#002b36, #839496, #002b36, #0d4350, #839496
> rofi.color-placeholder:	#000000
> rofi.color-text:	#000000
> rofi.color-urgent:	#a83838, #ffffff, #eee8d5, #a83838, #ffffff
> rofi.color-window:	#002b36, #255662, #0d4350
> rofi.font:	xos4Terminus 12
> rofi.m:	-1
> rofi.placeholder-color:	#000000
> rofi.text-color:	#000000
> ```

You are using it wrong, please see xrdb manpage.

**Reply by DaveDavenport** [reply]
> I can't hide the scrollbar though. I added `enabled: false;` to this part:
> 
> ```
> scrollbar {
>     enabled: false;
> }
> ```
> 
> It didnt' work though? What am I doing wrong?

see rofi-theme manpage.   there is a option on the listview (scrollbar is part of listview).

**Reply by Hund** [reply]
Oh. It was `load`, not `merge`. I guess I've been doing it wrong for a long time then. :)

I have read the manual for rofi-theme, but I can't see anything relevant there.

**Reply by DaveDavenport** [reply]
```
 listview:.IP • 2
       columns:         integer Number of columns to show (at least 1)

       • fixed-height:    boolean Always show lines rows, even if fewer elements are available.

       • dynamic:         boolean True if the size should change when filtering the list, False if it should keep the origi‐
         nal height.

       • scrollbar:       boolean If the scrollbar should be enabled/disabled.
```
so
```css
listview { scrollbar: false; }
```

**Reply by Hund** [reply]
Thank you! :)
